internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoint_service_serverless.go

```diff
@@ -62,48 +62,43 @@ func DataSource() *schema.Resource {
 	}
 }
 
-func dataSourceMongoDBAtlasPrivateEndpointServiceServerlessLinkRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	// Get client connection.
-	conn := meta.(*config.MongoDBClient).Atlas
+func dataSourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
 
 	projectID := d.Get("project_id").(string)
 	instanceName := d.Get("instance_name").(string)
 	endpointID := d.Get("endpoint_id").(string)
 
-	serviceEndpoint, _, err := conn.ServerlessPrivateEndpoints.Get(ctx, projectID, instanceName, endpointID)
+	serviceEndpoint, _, err := connV2.ServerlessPrivateEndpointsApi.GetServerlessPrivateEndpoint(ctx, projectID, instanceName, endpointID).Execute()
 	if err != nil {
 		return diag.FromErr(fmt.Errorf(privatelinkendpointservice.ErrorServiceEndpointRead, endpointID, err))
 	}
 
-	if err := d.Set("error_message", serviceEndpoint.ErrorMessage); err != nil {
+	if err := d.Set("error_message", serviceEndpoint.GetErrorMessage()); err != nil {
 		return diag.FromErr(fmt.Errorf(privatelinkendpointservice.ErrorEndpointSetting, "error_message", endpointID, err))
 	}
 
-	if err := d.Set("status", serviceEndpoint.Status); err != nil {
+	if err := d.Set("status", serviceEndpoint.GetStatus()); err != nil {
 		return diag.FromErr(fmt.Errorf(privatelinkendpointservice.ErrorEndpointSetting, "status", endpointID, err))
 	}
 
-	if err := d.Set("comment", serviceEndpoint.Comment); err != nil {
+	if err := d.Set("comment", serviceEndpoint.GetComment()); err != nil {
 		return diag.FromErr(fmt.Errorf(privatelinkendpointservice.ErrorEndpointSetting, "comment", endpointID, err))
 	}
 
```

Comment:
no ErrorMessage?

Reply:
Not sure I understand what you mean, can you explain it?

Reply:
if err := d.Set("error_message", serviceEndpoint.ErrorMessage); err != nil { ...

was deleted but not replaced

Reply:
Now I understand what you mean. Error message was duplicated here so I deleted one of them